src/gei/Commands/MigrateRepo/MigrateRepoCommandHandler.cs

```diff
@@ -69,7 +69,7 @@ public async Task Handle(MigrateRepoCommandArgs args)
 
         if (args.GhesApiUrl.HasValue())
         {
-            ValidateGHESOptions(args, blobCredentialsRequired);
+            ValidateGithubOptions(args, blobCredentialsRequired);
```

Comment:
You're still only validating the upload options if the `--ghes-api-url` is set! if the `--git-archive-path` is set the upload options need to be validated too. So to be clear this is what I am suggesting:

```C#
if (args.GhesApiUrl.HasValue() || args.GitArchivePath.HasValue())
{
    ValidateUploadOptions(args, blobCredentialsRequired);
}
```

There should also be unit tests for the migrate repo command to make sure the validation logic kicks in in both scenarios. There are example unit tests already for GHES path, you just need to add new tests for the archive path scenario.

also one extra thins that I just noticed is in case `--git-archive-path` is set `blobCredentialRequired` should be `true` so on line 68 we don't need to do the extra check if `--ghes-api-url` is not set:

```c#
var blobCredentialsRequired = args.GitArchivePath.HasValue() || await _ghesVersionChecker.AreBlobCredentialsRequired(args.GhesApiUrl);
```

Reply:
addressed in 0d0c8c94e2172ebef37ff63b0493f94c80575912